core/app/models/spree/order.rb

```diff
@@ -694,6 +694,10 @@ def covered_by_store_credit?
     end
     alias_method :covered_by_store_credit, :covered_by_store_credit?
 
+    def covered_by_valid_payments?
+      return payments.where(state: %w(checkout pending)).sum(:amount) >= total
```

Comment:
`return` not needed here.

Is there a pre-defined scope on payments we can use instead?